app/forms.py

```diff
@@ -0,0 +1,45 @@
+from django import forms
+from app.models.user import User
+from django.core.exceptions import ValidationError
+from django.contrib.auth import authenticate
+
+
+class SignUpForm(forms.ModelForm):
+    email = forms.EmailField(required=True)
+    password = forms.CharField(widget=forms.PasswordInput)
+    class Meta:
+        model = User
+        fields = ('email', 'passwrod')
+        widgets = {
+            'email' : forms.TextInput(attrs={'class': 'form_control'}),
+            'password' : forms.TextInput(attrs={'class': 'form_control'}),
+        }
+    def verify_email(self):
+        email = self.cleaned_data.get('email')
+        if User.objects.filter(email=email).exists():
+            raise ValidationError("Email is already in use.")
+        return email
+
+    def save(self, commit=True):
+        email = self.cleaned_data.get('email')
+        password = self.cleaned_data.get('password')
+        self.verify_email()
+        user = User(username=email,email=email)
+        user.set_password(password)
+        if commit:
+            user.save()
+        return user
+
+class LoginForm(forms.Form):
+    email = forms.EmailField(required=True)
+    password = forms.CharField(widget=forms.PasswordInput)
+
+    def clean(self):
+        cleaned_data = super(LoginForm, self).clean()
+        email = cleaned_data.get('email')
+        password = cleaned_data.get('password')
+        if email and password:
+            user = authenticate(username=email, password=password)
```

Comment:
we hit authentication twice 
one here and another in view function to fix it we need to remove it from here and add error message using add_error
check this https://docs.djangoproject.com/en/4.2/ref/forms/validation/#cleaning-a-specific-field-attribute
to know how to add error message to form